### PR TITLE
Tweak: Kit-Library - Create fallback url [ED-3544]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: core/app/modules/kit-library/connect/kit-library.php [hunk 1]
```diff
@@ -32,7 +32,10 @@ public function download_link( $id ) {
 	}
 
 	protected function get_api_url() {
```
Comment (matipojo) [inline core/app/modules/kit-library/connect/kit-library.php:34]:
```suggestion
	protected function get_api_urls() {
```

File: core/app/modules/kit-library/data/exceptions/wp-error-exception.php [hunk 2]
```diff
@@ -13,7 +13,10 @@ class Wp_Error_Exception extends \Exception {
 	protected $error;
 
 	public function __construct( \WP_Error $error ) {
-		parent::__construct( $error->get_error_message(), $error->get_error_code() );
+		$code = (int) $error->get_error_code();
+		$code = 0 === $code ? 500 : $code;
```
Comment (matipojo) [inline core/app/modules/kit-library/data/exceptions/wp-error-exception.php:17]: Explain

File: core/utils/http.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	 * @return bool
+	 */
+	private function is_successful_response( $response ) {
+		if ( is_wp_error( $response ) ) {
+			return false;
+		}
+
+		$response_code = (int) wp_remote_retrieve_response_code( $response );
+
+		if ( ! $response_code || 500 === $response_code || 404 === $response_code ) {
```
Comment (matipojo) [inline core/utils/http.php:44]: in_array( [0, 404, 500])

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/utils/http.php (hunk 3 above)
Comment (Nevoss) [reply core/utils/http.php:44]: Done

File: core/app/modules/kit-library/data/exceptions/wp-error-exception.php (hunk 2 above)
Comment (Nevoss) [reply core/app/modules/kit-library/data/exceptions/wp-error-exception.php:17]: Done

File: core/app/modules/kit-library/connect/kit-library.php (hunk 1 above)
Comment (Nevoss) [reply core/app/modules/kit-library/connect/kit-library.php:34]: Cannot change, extends from base.

**Review by matipojo — APPROVED** [review]

**Pull request merged**